pkg/sql/sqlstats/persistedsqlstats: BenchmarkSQLStatsFlush failed

pkg/sql/sqlstats/persistedsqlstats.BenchmarkSQLStatsFlush [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20353368?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/20353368?buildTab=artifacts#./artifacts/microbench/experiment) on master @ [8f5ae52a8f9a419efd0c4ab26464ff9b265ad1f3](https://github.com/cockroachdb/cockroach/commits/8f5ae52a8f9a419efd0c4ab26464ff9b265ad1f3):


```
        	            				pkg/testutils/subtest.go:23
        	            				GOROOT/src/testing/benchmark.go:219
        	            				GOROOT/src/testing/benchmark.go:245
        	            				src/runtime/asm_amd64.s:1700
        	Error:      	"517" is not greater than or equal to "1000"
        	Test:       	BenchmarkSQLStatsFlush/clusterSize=1/fingerprintCardinality=1000
--- FAIL: BenchmarkSQLStatsFlush/clusterSize=1/fingerprintCardinality=1000
BenchmarkSQLStatsFlush/clusterSize=1/fingerprintCardinality=7000
BenchmarkSQLStatsFlush/clusterSize=1/fingerprintCardinality=7000-32       	       1	10158185744 ns/op	5522290216 B/op	30453598 allocs/op
--- FAIL: BenchmarkSQLStatsFlush/clusterSize=1
BenchmarkSQLStatsFlush/clusterSize=3
BenchmarkSQLStatsFlush/clusterSize=3/fingerprintCardinality=10
    flush_test.go:1419: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/flush_test.go:1419
        	            				pkg/testutils/subtest.go:23
        	            				GOROOT/src/testing/benchmark.go:219
        	            				GOROOT/src/testing/benchmark.go:354
        	            				src/runtime/asm_amd64.s:1700
        	Error:      	"0" is not greater than or equal to "10"
        	Test:       	BenchmarkSQLStatsFlush/clusterSize=3/fingerprintCardinality=10
--- FAIL: BenchmarkSQLStatsFlush/clusterSize=3/fingerprintCardinality=10-32
BenchmarkSQLStatsFlush/clusterSize=3/fingerprintCardinality=100
    flush_test.go:1419: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/flush_test.go:1419
        	            				pkg/testutils/subtest.go:23
        	            				GOROOT/src/testing/benchmark.go:219
        	            				GOROOT/src/testing/benchmark.go:354
        	            				src/runtime/asm_amd64.s:1700
        	Error:      	"79" is not greater than or equal to "100"
        	Test:       	BenchmarkSQLStatsFlush/clusterSize=3/fingerprintCardinality=100
--- FAIL: BenchmarkSQLStatsFlush/clusterSize=3/fingerprintCardinality=100-32
BenchmarkSQLStatsFlush/clusterSize=3/fingerprintCardinality=1000
BenchmarkSQLStatsFlush/clusterSize=3/fingerprintCardinality=1000-32       	       1	5616929779 ns/op	4723191824 B/op	18554295 allocs/op
BenchmarkSQLStatsFlush/clusterSize=3/fingerprintCardinality=7000
BenchmarkSQLStatsFlush/clusterSize=3/fingerprintCardinality=7000-32       	       1	39583392921 ns/op	26609386416 B/op	124403138 allocs/op
--- FAIL: BenchmarkSQLStatsFlush/clusterSize=3
BenchmarkSQLStatsFlush/clusterSize=6
BenchmarkSQLStatsFlush/clusterSize=6/fingerprintCardinality=10
BenchmarkSQLStatsFlush/clusterSize=6/fingerprintCardinality=10-32         	      14	  73440767 ns/op	112437326 B/op	  357238 allocs/op
BenchmarkSQLStatsFlush/clusterSize=6/fingerprintCardinality=100
BenchmarkSQLStatsFlush/clusterSize=6/fingerprintCardinality=100-32        	       1	1205286403 ns/op	1292997192 B/op	 4201043 allocs/op
BenchmarkSQLStatsFlush/clusterSize=6/fingerprintCardinality=1000
BenchmarkSQLStatsFlush/clusterSize=6/fingerprintCardinality=1000-32       	       1	11156605640 ns/op	12122678184 B/op	40754836 allocs/op
BenchmarkSQLStatsFlush/clusterSize=6/fingerprintCardinality=7000
BenchmarkSQLStatsFlush/clusterSize=6/fingerprintCardinality=7000-32       	       1	83878899630 ns/op	66407910216 B/op	287065649 allocs/op
    flush_test.go:1426: -- test log scope end --
--- FAIL: BenchmarkSQLStatsFlush
FAIL

test logs left over in: /tmp/logBenchmarkSQLStatsFlush3812393293
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*BenchmarkSQLStatsFlush.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53573

--- a/pkg/sql/sqlstats/persistedsqlstats/flush_test.go
+++ b/pkg/sql/sqlstats/persistedsqlstats/flush_test.go
@@ -1354,7 +1354,8 @@ func BenchmarkSQLStatsFlush(b *testing.B) {
 			ServerArgs: base.TestServerArgs{
 				Knobs: base.TestingKnobs{
 					SQLStatsKnobs: &sqlstats.TestingKnobs{
-						StubTimeNow: fakeTime.Now,
+						StubTimeNow:         fakeTime.Now,
+						SynchronousSQLStats: true,
 					},
 				},
 			},
@@ -1386,7 +1387,7 @@ func BenchmarkSQLStatsFlush(b *testing.B) {
 
 		gen := genPermutations()
 
-		testutils.RunValues(b, "fingerprintCardinality", []int64{10, 100, 1000, 7000}, func(b *testing.B, uniqueFingerprintCount int64) {
+		testutils.RunValues(b, "fingerprintCardinality", []int64{10, 100, 1000}, func(b *testing.B, uniqueFingerprintCount int64) {
 			ctx := context.Background()
 			for iter := 0; iter < b.N; iter++ {
 				for i := int64(0); i < uniqueFingerprintCount; i++ {

